core/src/main/java/org/eclipse/microprofile/reactive/streams/core/PublisherBuilderImpl.java

```diff
@@ -0,0 +1,242 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.eclipse.microprofile.reactive.streams.core;
```

Comment:
Wondering about the package name? 
Should it be `impl`?

Reply:
I don't have a strong opinion. The reason I didn't call it `impl` is that it's not a full implementation of the spec, it's just an implementation of the builder API, the core part of the spec. Plus, calling it `impl` might give the impression that this is "the implementation", which isn't supposed to exist. A better description of this might be core utilities for implementing it.